client/rest-high-level/src/test/java/org/elasticsearch/client/watcher/WatchStatusTests.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.watcher;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.XContentTestUtils;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.io.IOException;
+import java.util.function.Predicate;
+
+public class WatchStatusTests extends ESTestCase {
+
+    public void testBasicParsing() throws IOException {
```

Comment:
In order to sanity check this PR, I manually added some basic parsing tests. One other option would be to keep the `toXContent` logic, and create a `AbstractXContentTestCase`. I decided against that approach because there was quite a bit of logic associated with `toXContent`, and this isn't the test direction we're ultimately aiming for.

Reply:
I think you can avoid that by overriding `AbstractXContentTestCase#assertToXContentEquivalence`?  I think it's worth using `AbstractXContentTestCase` here, it's going to be much more thorough than hand-rolling parsing tests.

Reply:
we need to come to consensus on that, since we wont actually have the toXContent on a lot of classes, and i dont know if i want to add it solely for completing a test. Also, testing that the class can ser/deser the bytes is one thing, but testing that a server side response ser's into something that the client can deser is much more valuable, since thats the way we know the client actually works. After support summit ill have a few brain cells to devote to this as Id like to get an answer out sooner than later.

Reply:
Thanks everyone for taking a look. I purposefully removed the `toXContent` method and stream serialization methods (in the commit “Remove logic related to writing xContent and stream serialization”) because in the high-level REST client, response classes are only ever deserialized from xContent. As @hub-cap mentioned, we expect this to be the case for most of the response classes.

In tests like `AbstractXContentTestCase`, we test parsing by creating an object, generating xContent, then parsing it back to an object and checking equality. So because we no longer have a `toXContent` method, this testing strategy doesn’t apply. My understanding was that we did not want to keep around `toXContent` just to help with testing, and would prefer to move in a direction where we’re testing against a live server (or some kind of API spec?)

@hub-cap do you have thoughts about the right testing approach for this PR? It would be nice make progress on getting this merged (even if the testing approach is temporary), so others can continue their watcher HLRC work.

Reply:
I need to sit down to make a approach, ill finally have time early next week. Id prefer the manual parsing in order to test the fromXContents, for now.

Reply:
I clarified with @hub-cap and he thinks we should stick with this testing approach for this PR + proceed with the code review (or as he says, :shipit: :) )

Reply:
There are already a couple of actions implemented. For example this one which is the delete watcher action:

https://github.com/elastic/elasticsearch/pull/32337

Responses extends the abstract class ActionResponse which seems to require serialisation/deserialisation. The responses classes implement ToXContentObject as well  so no sure if for consistency we should do the same.

Reply:
Right, but looking at a more recent xpack HLRC PR (#32332), I saw that the responses no longer contained the extraneous serialization/ deserialization logic (see `PutUserResponse`). I assumed that this was the new preferred way, especially now that we are moving the responses into their own lightweight module. Tagging @hub-cap in case my assumptions are off!

Reply:
@jtibshirani is correct

Reply:
This other approach looks fine to me as well. Thanks for the clarification!